doc/indexing.rst

```diff
@@ -353,67 +331,129 @@ labels of the 1st and 2nd dimension, respectively. You must specify all
+.. _advanced_indexing:
 
-.. _copies vs views:
+Basic and Advanced Indexing
+---------------------------
 
-Copies vs. views
-----------------
+As similar to numpy's nd-array, xarray supports two types of indexing,
+`basic- and advanced-indexing`__.
+However, our indexing rule differs from numpy.
 
-Whether array indexing returns a view or a copy of the underlying
-data depends on the nature of the labels. For positional (integer)
-indexing, xarray follows the same rules as NumPy:
+__ https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html
 
-* Positional indexing with only integers and slices returns a view.
-* Positional indexing with arrays or lists returns a copy.
 
-The rules for label based indexing are more complex:
+Our indexing is basically orthogonal, i.e.
+if you pass multiple integer sequences to an array, they work independently
+along each dimension (similar to the way vector subscripts work in fortran).
 
-* Label-based indexing with only slices returns a view.
-* Label-based indexing with arrays returns a copy.
-* Label-based indexing with scalars returns a view or a copy, depending
-  upon if the corresponding positional indexer can be represented as an
-  integer or a slice object. The exact rules are determined by pandas.
+.. ipython:: python
 
-Whether data is a copy or a view is more predictable in xarray than in pandas, so
-unlike pandas, xarray does not produce `SettingWithCopy warnings`_. However, you
-should still avoid assignment with chained indexing.
+    da = xr.DataArray(np.arange(12).reshape((3, 4)), dims=['x', 'y'],
+                      coords={'x': [0, 1, 2], 'y': ['a', 'b', 'c', 'd']})
+    da
+    da[[0, 1], [1, 1]]
+    # Sequential indexing gives the same result.
+    da[[0, 1], [1, 1]] == da[[0, 1]][:, [1, 1]]
 
-.. _SettingWithCopy warnings: http://pandas.pydata.org/pandas-docs/stable/indexing.html#returning-a-view-versus-a-copy
 
-.. _orthogonal:
+In order to make more advanced indexing, you can supply
+:py:meth:`~xarray.DataArray` as indexers.
+In this case, the dimension of the resultant array is determined
+by the indexers' dimension names,
+
+.. ipython:: python
+
+    ind_x = xr.DataArray([0, 1], dims=['x'])
+    ind_y = xr.DataArray([0, 1], dims=['y'])
+    da[ind_x, ind_y]  # orthogonal indexing
+    da[ind_x, ind_x]  # vectorized indexing
+
+If you just provide slices or sequences, which do not have named-dimensions,
+they will be understood as the same dimension which is indexed along.
+
+.. ipython:: python
+
+    # Because [0, 1] is used to index along dimension 'x',
+    # it is assumed to have dimension 'x'
+    da[[0, 1], ind_x]
+
+
+Furthermore, you can use multi-dimensional :py:meth:`~xarray.DataArray`
+as indexers, where the resultant array dimension is also determined by
+indexers' dimension,
+
+.. ipython:: python
+
+    ind = xr.DataArray([[0, 1], [0, 1]], dims=['a', 'b'])
+    da[ind]
+
+To summarize, our advanced indexing is based on our broadcasting scheme.
```

Comment:
We should also list the full set of rules we use for indexing separate from examples for reference purposes. Maybe this should be a new section at the bottom.